kfbot-refactor-phase2/src/main/java/com/twitchbotx/bot/handlers/CountHandler.java

```diff
@@ -184,57 +184,77 @@ public static String handleSubMessage(
-                        response = guiHandler.bot.getStore().getConfiguration().subPrimeReply;
-                        response = response.replace("%user", subDisplayName).replace("%months", subMonths);
-                    } else {
-                        // find a replace variables %user %months %tier
-                        response = guiHandler.bot.getStore().getConfiguration().subNormalReply;
-                        response = response.replace("%user", subDisplayName).replace("%months", subMonths).replace("%tier", tier);
-                    }
-                }
+
+            String response = "";
+
+            // deal with mass gift sub
+            if (massGifted) {
+                // find a replace variables %user %gifts %tier
+                response = guiHandler.bot.getStore().getConfiguration().subMassGiftReply;
+                response = response.replace("%user", subDisplayName).replace("%gifts", String.valueOf(giftAmount)).replace("%tier", tier);
+                
+                // bundle and send reply to messenger and event list
+                sendEvent(response);
+                return response;
             }
+
+            // massGifted can only be false if it reaches this point
+
             // deal with single gifted sub
-            if (!massGifted && subGift) {
+            if (subGift) {
                 // find a replace variables %user %recipient %tier
                 response = guiHandler.bot.getStore().getConfiguration().subSingleGiftReply;
                 response = response.replace("%user", subDisplayName).replace("%recipient", giftRecip).replace("%tier", tier);
+                
+                // bundle and send reply to messenger and event list
+                sendEvent(response);
+                return response;
             }
-            if (massGifted) {
-                // find a replace variables %user %gifts %tier
-                response = guiHandler.bot.getStore().getConfiguration().subMassGiftReply;
-                response = response.replace("%user", subDisplayName).replace("%gifts", String.valueOf(giftAmount)).replace("%tier", tier);
+
+            // the only way to reach this level is if massGifted and subGift are false
+
+            // new prime sub
+            if (Integer.parseInt(subMonths) < 2 && prime) {
+                // find and replace variable %user %tier
+                response = guiHandler.bot.getStore().getConfiguration().subNewPrimeReply;
+                response = response.replace("%user", subDisplayName).replace("%tier", tier);
+            } 
+
+            // new sub
+            else if (Integer.parseInt(subMonths) < 2) {
+                // find and replace variable %user %tier
+                response = guiHandler.bot.getStore().getConfiguration().subNewNormalReply;
+                response = response.replace("%user", subDisplayName).replace("%tier", tier);
+            } 
+
+            // prime sub
```

Comment:
Should specify prime resub here
```suggestion
            // prime resub
```